utils_linux.go

```diff
@@ -171,6 +171,9 @@ func createPidFile(path string, process *libcontainer.Process) error {
 	if err != nil {
 		return err
 	}
+	if _, err := os.Stat(path); err == nil {
+		return fmt.Errorf("file %s already exists", path)
```

Comment:
I'm a bit worried that is going to be a moving target we have to remember to update each time we add a new flag. Maybe if we check the d_type and if it's not a normal file we error out (the `Rename` would cause it to be overridden anyway which is counter-intuitive if you want the `pid` to be written to a FIFO for example).